ga4gh/datamodel/variants.py

```diff
@@ -787,9 +787,20 @@ def buildMetadata(
         return ret
 
     def isAnnotated(self, path):
+        # assumes that all files in the directory look like the first
+        return self.hasAnnField(path) or self.hasConsequenceField(path)
+
+    def hasAnnField(self, path):
+        pysamreader = self.openFile(
+            glob.glob(os.path.join(path, "*.vcf.gz"))[0])
+        items = [x[0] for x in pysamreader.header.info.items()]
+        return 'ANN' in items
+
+    def hasConsequenceField(self, path):
         pysamreader = self.openFile(
             glob.glob(os.path.join(path, "*.vcf.gz"))[0])
```

Comment:
I moved the repeated code into a separate function. When adding a variant set you are adding a directory, and there are no guarantees for the name. https://github.com/ga4gh/server/blob/master/ga4gh/datamodel/datasets.py#L219